superduperdb/ext/sklearn/model.py

```diff
@@ -5,99 +5,90 @@
 from sklearn.base import BaseEstimator
 from tqdm import tqdm
 
-from superduperdb.backends.base.query import Select
 from superduperdb.backends.query_dataset import QueryDataset
 from superduperdb.base.datalayer import Datalayer
 from superduperdb.components.datatype import DataType, pickle_serializer
-from superduperdb.components.metric import Metric
 from superduperdb.components.model import (
-    Mapping,
+    ModelInputType,
+    Signature,
+    Trainer,
     _Fittable,
     _Predictor,
-    _TrainingConfiguration,
+    _Validator,
 )
 from superduperdb.jobs.job import Job
 
 
-def _get_data_from_query(
-    select: Select,
-    X: str,
-    db: Datalayer,
-    y: t.Optional[str] = None,
-    y_preprocess: t.Optional[t.Callable] = None,
-    preprocess: t.Optional[t.Callable] = None,
-):
-    if y is None:
-        data = QueryDataset(
-            select=select,
-            mapping=Mapping([X], signature='singleton'),
-            transform=preprocess,
-            db=db,
-        )
-    else:
-        y_preprocess = y_preprocess or (lambda x: x)
-        preprocess = preprocess or (lambda x: x)
-        data = QueryDataset(
-            select=select,
-            mapping=Mapping([X, y], signature='*args'),
-            transform=lambda x, y: (preprocess(x), y_preprocess(y)),
-            db=db,
-        )
-
-    rows = []
-    for i in tqdm(range(len(data))):
-        rows.append(data[i])
-    if y is not None:
-        X_arr = [x[0] for x in rows]
-        y_arr = [x[1] for x in rows]
-    else:
-        X_arr = rows
-
-    if isinstance(X[0], numpy.ndarray):
-        X_arr = numpy.stack(X_arr)
-    if y is not None:
-        y_arr = [r[1] for r in rows]
-        if isinstance(y_arr[0], numpy.ndarray):
-            y_arr = numpy.stack(y_arr)
-        return X_arr, y_arr
-    return X_arr, None
-
-
-@dc.dataclass
-class SklearnTrainingConfiguration(_TrainingConfiguration):
+@dc.dataclass(kw_only=True)
+class SklearnTrainer(Trainer):
     fit_params: t.Dict = dc.field(default_factory=dict)
     predict_params: t.Dict = dc.field(default_factory=dict)
     y_preprocess: t.Optional[t.Callable] = None
 
+    @staticmethod
+    def _get_data_from_dataset(dataset, X: ModelInputType):
+        assert isinstance(X, (tuple, list, str))
+        if isinstance(X, (tuple, list)):
+            assert len(X) == 2
+
+        rows = []
+        for i in tqdm(range(len(dataset))):
```

Comment:
is this to show user len of dataset via tqdm?

Reply:
It's to show progress of loading data.